src/transport_ofi.c

```diff
@@ -678,15 +678,17 @@ int allocate_recv_cntr_mr(void)
     /* Register separate data and heap segments using keys 0 and 1,
      * respectively.  In MR_BASIC_MODE, the keys are ignored and selected by
      * the provider. */
+    uint64_t key = 1;
```

Comment:
These key values are ignored by the provider if it requires `FI_MR_PROV_KEY` right?

I'm just checking whether or not we need a build flag for this.  My understanding is that this is a mode some providers require but SOS should support either way (which is why I hope the key value is ignored here if the mode is MR_BASIC/MR_PROV_KEY).  Is that the case?

Reply:
Yes, the provider is going to ignore the requested key, if SOS requests for `FI_MR_PROV_KEY` and the OFI provider also supports it. In such case, it will use its own key and we can extract that key with `fi_mr_key`.